### PR TITLE
test: Improve unit test coverage for status

### PR DESCRIPTION
**Pull request by rakesh-garimella**
## Description

Changes proposed in this pull request (what was done and why):

- The status file for different pipelines does not have unit test coverage for tlsCertificate verification add the tests to improve the coverage

Changes refer to particular issues, PRs or documents:

- 

## Traceability
- [ ] The PR is linked to a GitHub issue.
- [ ] New features have a milestone set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] Adjusted the documentation if the change is user-facing.
- [x] The feature is unit-tested
- [ ] The feature is e2e-tested

### CODE REVIEW

File: internal/testutils/pipline_opts.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	}
+}
+
+type CustomHostOption func() string
+
+func CustomHost(host string) CustomHostOption {
+	return func() string {
+		return host
+	}
+}
```
Comment (skhalash) [inline internal/testutils/pipline_opts.go:103]: What is this for? It's not doing anything, is it?

File: internal/testutils/trace_pipeline_builder.go [hunk 2]
```diff
@@ -32,19 +32,15 @@ func (b *TracePipelineBuilder) WithName(name string) *TracePipelineBuilder {
...
-	b.basicAuthUser = user
-	b.basicAuthPassword = password
+func (b *TracePipelineBuilder) WithStatusCondition(cond metav1.Condition) *TracePipelineBuilder {
+	b.statusConditions = append(b.statusConditions, cond)
 	return b
 }
 
-func (b *TracePipelineBuilder) WithStatusCondition(cond metav1.Condition) *TracePipelineBuilder {
-	b.statusConditions = append(b.statusConditions, cond)
+func (b *TracePipelineBuilder) OtlpOutput(opts ...OTLPOutputOption) *TracePipelineBuilder {
```
Comment (skhalash) [inline internal/testutils/trace_pipeline_builder.go:40]: In the opts you are using OTLP, but here it's Otlp. Maybe let's unify it and use OTLP everywhere?

File: internal/testutils/log_pipeline_builder.go [hunk 3]
```diff
@@ -13,14 +13,28 @@ import (
...
+		},
+	}
+}
+
+func (b *LogPipelineBuilder) WithLoki() *LogPipelineBuilder {
+	b.lokiOutput = &telemetryv1alpha1.LokiOutput{
+		URL: telemetryv1alpha1.ValueType{Value: "https://localhost:3100"},
 	}
+	return b
 }
```
Comment (skhalash) [inline internal/testutils/log_pipeline_builder.go:38]: Why do we have WithLoki and LokiOutput?

File: internal/testutils/log_pipeline_builder.go [hunk 4]
```diff
@@ -33,18 +47,53 @@ func (b *LogPipelineBuilder) WithStatusCondition(cond metav1.Condition) *LogPipe
...
+	for _, opt := range opts {
+		opt(b.lokiOutput)
+	}
+	return b
+}
+
+func (b *LogPipelineBuilder) CustomOutput(opt CustomHostOption) *LogPipelineBuilder {
+	b.customOutput = opt()
+	return b
+}
```
Comment (skhalash) [inline internal/testutils/log_pipeline_builder.go:67]: We don't really need any opts here. You can only provide it as a string
```suggestion
func (b *LogPipelineBuilder) CustomOutput(custom string) *LogPipelineBuilder {
	b.customOutput = custom
	return b
}
```

File: internal/testutils/metric_pipeline_builder.go
Comment (skhalash) [inline internal/testutils/metric_pipeline_builder.go:1]: Some builder methods have a With prefix, and some do not. Could you please unify it for all 3 pipeline types and add With everywhere?

**Review by skhalash — CHANGES_REQUESTED** [review]

File: internal/testutils/pipline_opts.go (hunk 1 above)
Comment (rakesh-garimella) [reply internal/testutils/pipline_opts.go:103]: removed

File: internal/testutils/log_pipeline_builder.go (hunk 3 above)
Comment (rakesh-garimella) [reply internal/testutils/log_pipeline_builder.go:38]: removed LokiOutput

File: internal/testutils/trace_pipeline_builder.go (hunk 2 above)
Comment (rakesh-garimella) [reply internal/testutils/trace_pipeline_builder.go:40]: done

**Pull request merged**